### PR TITLE
feat(footprints): layer footprint by drawn polugon 

### PR DESCRIPTION
**Pull request by alebinson**
Contains remarked section of implementations of discrete-layers geojson as is, current version shows discrete-layer bounding-box

### CODE REVIEW

File: src/discrete-layer/models/discreteLayersStore.ts [hunk 1]
```diff
@@ -45,14 +50,41 @@ export const discreteLayersStore = types
...
-          self.layersImages = result.map(item => ({...item, selected:false}));
+          self.layersImages = filterBySearchParams(result).map(item => ({...item, selected:false}));
         } catch (error) {
           console.error(error);
           self.state = ResponseState.ERROR;
         }
       }
     );
 
+    function filterBySearchParams(layers: ILayerImage[]): ILayerImage[] {
```
Comment (asafMasa) [inline src/discrete-layer/models/discreteLayersStore.ts:61]: add todo: to remove this method when we will get a real data

File: src/discrete-layer/components/map-container/layers-footprints.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  const [layersFootprints, setlayersFootprints] = useState<FeatureCollection>();
+
+  // REMARK: Layers footprint boundingboxes  
+  useEffect(() => {
+    if (discreteLayersStore.layersImages) {
+      const footprintsCollection: FeatureCollection = {
+        type: 'FeatureCollection',
+        features: []
+      }
+      const footPrintsFeaturesArray = discreteLayersStore.layersImages.map((layer) => {
```
Comment (asafMasa) [inline src/discrete-layer/components/map-container/layers-footprints.tsx:21]: footPrints -> footprints

File: src/discrete-layer/components/map-container/layers-footprints.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        let geometry: Geometry = layer.geojson as Geometry;
+        switch(layer.geojson?.type){
+          case 'Polygon':
+            geometry = (polygonToLine(geometry as Polygon) as Feature).geometry;
+            break;
+          case 'MultiPolygon':
+            //get bbox of feture, then convert it to polygon and finally get linestring of polygon
+            geometry = (polygonToLine(bboxPolygon(bbox(geometry))) as Feature).geometry;
```
Comment (asafMasa) [inline src/discrete-layer/components/map-container/layers-footprints.tsx:29]: image
why the conversion to BBOX when Turf can handle it?

File: src/discrete-layer/components/map-container/layers-footprints.tsx [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+            break;
+          default:
+            break;
+        }
+        
+        const footPrint: Feature = {
```
Comment (asafMasa) [inline src/discrete-layer/components/map-container/layers-footprints.tsx:35]: footPrint -> footprint

File: src/discrete-layer/components/map-container/layers-footprints.tsx [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+          },
+        }
+        return footPrint;
+      });
+      footprintsCollection.features.push(...footPrintsFeaturesArray);
+      setlayersFootprints(footprintsCollection);
+    }
+  }, [discreteLayersStore.layersImages]);
+
+  // REMARK: Layers feature footprint as is
```
Comment (asafMasa) [inline src/discrete-layer/components/map-container/layers-footprints.tsx:52]: remove REMARK?

File: src/discrete-layer/components/map-container/layers-footprints.tsx [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+  //     });
+  //     footprintsCollection.features.push(...footPrintsFeaturesArray);
+  //     setlayersFootprints(footprintsCollection);
+  //   }
+  // }, [discreteLayersStore.layersImages]);
+
+  return (
+    <CesiumGeojsonLayer
+      data={layersFootprints}
+      // markerColor={CesiumColor.RED}
```
Comment (asafMasa) [inline src/discrete-layer/components/map-container/layers-footprints.tsx:80]: remove comment?

File: src/discrete-layer/components/map-container/layers-footprints.tsx [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+      onLoad={(g): void => {
```
Comment (asafMasa) [inline src/discrete-layer/components/map-container/layers-footprints.tsx:81]: g -> use meaningful names

File: src/discrete-layer/components/map-container/layers-footprints.tsx [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+        
+        // REMARK: Unified boundingboxes of footprints
+        g.entities.values.forEach(item => {
+          // @ts-ignore
+          item.polyline.width = 6.0;
```
Comment (asafMasa) [inline src/discrete-layer/components/map-container/layers-footprints.tsx:86]: fix eslint

File: src/discrete-layer/components/map-container/layers-footprints.tsx [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+          // @ts-ignore
+          item.polyline.material = CesiumColor.RED;
+        });
+
+        // REMARK: footprints as is
```
Comment (asafMasa) [inline src/discrete-layer/components/map-container/layers-footprints.tsx:91]: remove comments?

File: src/discrete-layer/models/discreteLayersStore.ts [hunk 10]
```diff
@@ -45,14 +50,41 @@ export const discreteLayersStore = types
           // eslint-disable-next-line @typescript-eslint/ban-ts-comment
           // @ts-ignore
           const result = yield  Promise.resolve(MOCK_DATA_IMAGERY_LAYERS_ISRAEL);
-          self.layersImages = result.map(item => ({...item, selected:false}));
+          self.layersImages = filterBySearchParams(result).map(item => ({...item, selected:false}));
```
Comment (asafMasa) [inline src/discrete-layer/models/discreteLayersStore.ts:53]: createMockData - not in use fix lint
image

**Review by asafMasa — CHANGES_REQUESTED** [review]
👍 good job

File: src/discrete-layer/models/discreteLayersStore.ts (hunk 1 above)
Comment (alebinson) [reply src/discrete-layer/models/discreteLayersStore.ts:61]: Done

File: src/discrete-layer/components/map-container/layers-footprints.tsx (hunk 2 above)
Comment (alebinson) [reply src/discrete-layer/components/map-container/layers-footprints.tsx:21]: Done

File: src/discrete-layer/components/map-container/layers-footprints.tsx (hunk 3 above)
Comment (alebinson) [reply src/discrete-layer/components/map-container/layers-footprints.tsx:29]: For this case I need only outer Feature which is presented as single LineString geojson Feature.

In this file present two kind of visualizations: **outer BBOX** (Feature<LineString>) and **geoJSON feature as is** only one of them should be chosen @shaziri want a geoJSON as is with different fill colors.

Think at this time no action item, code remained unchanged.

File: src/discrete-layer/components/map-container/layers-footprints.tsx (hunk 4 above)
Comment (alebinson) [reply src/discrete-layer/components/map-container/layers-footprints.tsx:35]: Done

File: src/discrete-layer/components/map-container/layers-footprints.tsx (hunk 5 above)
Comment (alebinson) [reply src/discrete-layer/components/map-container/layers-footprints.tsx:52]: It's a second **geojson as is** implementation

Remained unchanged

File: src/discrete-layer/components/map-container/layers-footprints.tsx (hunk 6 above)
Comment (alebinson) [reply src/discrete-layer/components/map-container/layers-footprints.tsx:80]: Done

File: src/discrete-layer/components/map-container/layers-footprints.tsx (hunk 7 above)
Comment (alebinson) [reply src/discrete-layer/components/map-container/layers-footprints.tsx:81]: Done

File: src/discrete-layer/components/map-container/layers-footprints.tsx (hunk 9 above)
Comment (alebinson) [reply src/discrete-layer/components/map-container/layers-footprints.tsx:91]: It's a second version of **geoJason as is**

code remained unchanged

File: src/discrete-layer/models/discreteLayersStore.ts (hunk 10 above)
Comment (alebinson) [reply src/discrete-layer/models/discreteLayersStore.ts:53]: It's a generic way to mimic data bulks, generates as much as defined entries.
remain code unchanged.

Till we in massive development and not all decisions are taken I prefer to leave code with my comments and remarks.

File: src/discrete-layer/components/map-container/layers-footprints.tsx (hunk 8 above)
Comment (alebinson) [reply src/discrete-layer/components/map-container/layers-footprints.tsx:86]: Partially fixed,
There is am issue that deals with it https://github.com/CesiumGS/cesium/issues/8898

**Review by asafMasa — APPROVED** [review]

**Pull request merged**